src/nlp_expr.jl

```diff
@@ -368,16 +368,20 @@ for f in (:+, :-, :*, :^, :/, :atan)
 end
 
 function _MA.operate!!(
-    ::typeof(_MA.add_mul),
+    op::_MA.AddSubMul,
     x::GenericNonlinearExpr,
-    y::AbstractJuMPScalar,
-)
+    args::Vararg{Any,N},
+) where {N}
     _throw_if_not_real(x)
     if x.head == :+
-        push!(x.args, y)
+        arg = *(args...)
+        if _MA.add_sub_op(op) != +
+            arg = _MA.add_sub_op(op)(arg)
```

Comment:
I really think we should avoid this. If the user writes `x + y - 2 * z`, we should make `-(+(x, y), *(2, z))`, not `+(x, y, -(*(2, z))). 

Julia has a parsing precedence order. Let's follow that where possible.

Reply:
I thought `flatten!` was doing that anyway